x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/enrollment/NodeEnrollmentAction.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.security.action.enrollment;
+
+import org.elasticsearch.action.ActionType;
+
+public final class NodeEnrollmentAction extends ActionType<NodeEnrollmentResponse> {
+
+    public static final String NAME = "cluster:admin/xpack/enroll";
```

Comment:
Barring the enroll credential, no one else should be able to invoke this API (because it is not useful and because you get the HTTP CA key and can impersonate the cluster).
I suggest we exclude this action from all the existing privileges; so that it has to be granted explicitly.

Reply:
yes! Thanks for calling me out on this. We have discussed this extensively and we don't want any other privilege to imply enroll / manage enrollment  or to give access to these actions. I'll move this over to a specific prefix, maybe simply `enrollment:enroll` for this

Reply:
@albertzaharovits how would you feel about putting this under `cluster:admin/xpack/security/enroll` and restricting this from `manage_security`, vs leaving this as is and excluding it from the `manage` cluster ? Or should we go all the way and exclude this from `all` too ?

Reply:
Please create an issue for this, since I understand this will be a follow-up.